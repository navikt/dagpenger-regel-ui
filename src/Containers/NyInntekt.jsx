import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { Form, withFormik } from 'formik';
import { Hovedknapp, Knapp } from 'nav-frontend-knapper';
import { Undertittel } from 'nav-frontend-typografi';
import InputField from '../Components/InputField';
import SelectField from '../Components/SelectField';
import { required } from '../Utils/validering';
import DatoLabel from '../Components/DatoLabel';
import { MMMM_YYYY_FORMAT } from '../Utils/datoFormat';
import { VerdikoderContext } from '../Context/Verdikoder';


const mapVerdikoder = typer => typer
  .map(navn => (<option value={navn} key={navn}>{navn}</option>));

const NyInntekt = (props) => {
  const {
    handleSubmit,
    isSubmitting,
    closeModal,
    virksomhet,
    isValid,
    dato,
  } = props;
  const verdikoder = useContext(VerdikoderContext);
  return (
    <Form onSubmit={handleSubmit}>
      <Undertittel>
        {`Legg til ny inntektspost for ${virksomhet.identifikator} i `}
        <DatoLabel dato={dato} datoFormat={MMMM_YYYY_FORMAT} />
      </Undertittel>

      <SelectField
        bredde="xl"
        label="Beskrivelse"
        selectValues={mapVerdikoder(verdikoder)}
        validate={required}
        name="verdikode"
        readOnly={false}
      />

      <InputField
        label="BelÃ¸p"
        name="beloep"
        validate={required}
        type="number"
        readOnly={false}
      />

      <div className="knapprad">
        <Hovedknapp
          htmlType="submit"
          onClick={handleSubmit}
          disabled={!isValid || isSubmitting}
        >
      Legg til
        </Hovedknapp>
        <Knapp
          htmlType="button"
          onClick={closeModal}
        >
      Avbryt
        </Knapp>
      </div>

    </Form>
  );
};

NyInntekt.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  virksomhet: PropTypes.shape().isRequired,
  dato: PropTypes.string.isRequired,
  closeModal: PropTypes.func.isRequired,
  isSubmitting: PropTypes.bool.isRequired,
  isValid: PropTypes.bool.isRequired,
};

export default withFormik({
  mapPropsToValues: props => ({
    verdikode: '',
    beloep: '0.00',
    dato: props.dato,
    virksomhet: props.virksomhet,
    opplysningspliktig: props.opplysningspliktig,
  }),

  handleSubmit: (values, { setSubmitting, props }) => {
    const { closeModal, arrayHelpers } = props;
    arrayHelpers.push({
      verdikode: values.verdikode,
      beloep: values.beloep,
      utbetaltIMaaned: values.dato,
      fordel: 'denne kommer',
      inntektskilde: 'dagpenger-regel-ui',
      inntektsperiodetype: 'Maaned',
      virksomhet: values.virksomhet,
      inntektsmottaker: arrayHelpers.form.values.inntekt.ident,
      opplysningspliktig: values.virksomhet,

    });
    setSubmitting(false);
    closeModal();
  },

  displayName: 'nyInntekt',

})(NyInntekt);
