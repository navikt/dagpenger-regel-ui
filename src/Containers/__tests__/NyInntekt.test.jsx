import React from 'react';
import renderer from 'react-test-renderer';
import NyInntekt from '../NyInntekt';

const arrayHelpers = {

};

const setModal = () => {

};

test('Skal vise ny inntektspost', () => {
  const component = renderer.create(
    <NyInntekt arbeidsgiver="9876543210" dato="2019-05" arrayHelpers={arrayHelpers} closeModal={() => setModal(false)} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
