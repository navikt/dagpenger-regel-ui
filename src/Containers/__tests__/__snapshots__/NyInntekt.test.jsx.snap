// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skal vise  1`] = `
Array [
  <h2
    className="typo-undertittel"
  >
    Legg til ny inntektspost for 9876543210 i 
    mai 2019
  </h2>,
  <div
    className="skjemaelement"
  >
    <label
      className="skjemaelement__label"
      htmlFor="beskrivelse"
    >
      <div
        className="flex label"
      >
        Beskrivelse
      </div>
    </label>
    <input
      className="skjemaelement__input input--fullbredde"
      id="beskrivelse"
      name="beskrivelse"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value=""
    />
    <div
      aria-live="assertive"
      role="alert"
    />
  </div>,
  <div
    className="skjemaelement"
  >
    <label
      className="skjemaelement__label"
      htmlFor="beloep"
    >
      <div
        className="flex label"
      >
        Beløp
      </div>
    </label>
    <input
      className="skjemaelement__input input--fullbredde"
      id="beloep"
      name="beloep"
      onBlur={[Function]}
      onChange={[Function]}
      type="number"
      value="0.00"
    />
    <div
      aria-live="assertive"
      role="alert"
    />
  </div>,
  <div
    className=""
    field={
      Object {
        "name": "nyInntektUtloeserArbeidsgiveravgift",
        "onBlur": [Function],
        "onChange": [Function],
        "value": undefined,
      }
    }
    form={
      Object {
        "dirty": false,
        "errors": Object {},
        "handleBlur": [Function],
        "handleChange": [Function],
        "handleReset": [Function],
        "handleSubmit": [Function],
        "initialValues": Object {
          "beloep": "0.00",
          "beskrivelse": "",
        },
        "isSubmitting": false,
        "isValid": false,
        "isValidating": false,
        "registerField": [Function],
        "resetForm": [Function],
        "setError": [Function],
        "setErrors": [Function],
        "setFieldError": [Function],
        "setFieldTouched": [Function],
        "setFieldValue": [Function],
        "setFormikState": [Function],
        "setStatus": [Function],
        "setSubmitting": [Function],
        "setTouched": [Function],
        "setValues": [Function],
        "status": undefined,
        "submitCount": 0,
        "submitForm": [Function],
        "touched": Object {},
        "unregisterField": [Function],
        "validateField": [Function],
        "validateForm": [Function],
        "validateOnBlur": true,
        "validateOnChange": true,
        "values": Object {
          "beloep": "0.00",
          "beskrivelse": "",
        },
      }
    }
  >
    <fieldset
      className="skjema__fieldset"
    >
      <legend
        className="skjema__legend"
      >
        Utløser arbeidsgiveravgift?
      </legend>
      <div
        className="skjemaelement skjemaelement--horisontal"
      >
        <input
          className="skjemaelement__input radioknapp"
          id="nyInntektJa"
          name="nyInntektUtloeserArbeidsgiveravgift"
          type="radio"
          value={true}
        />
        <label
          className="skjemaelement__label"
          htmlFor="nyInntektJa"
        >
          Ja
        </label>
      </div>
      <div
        className="skjemaelement skjemaelement--horisontal"
      >
        <input
          className="skjemaelement__input radioknapp"
          id="nyInntektNei"
          name="nyInntektUtloeserArbeidsgiveravgift"
          type="radio"
          value={false}
        />
        <label
          className="skjemaelement__label"
          htmlFor="nyInntektNei"
        >
          Nei
        </label>
      </div>
    </fieldset>
    <div
      aria-live="assertive"
      role="alert"
    />
  </div>,
  <button
    className="knapp knapp--hoved"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Legg til
  </button>,
  <button
    className="knapp knapp--standard"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Avbryt
  </button>,
]
`;

exports[`Skal vise ny inntektspost 1`] = `
Array [
  <h2
    className="typo-undertittel"
  >
    Legg til ny inntektspost for 9876543210 i 
    mai 2019
  </h2>,
  <div
    className="skjemaelement"
  >
    <label
      className="skjemaelement__label"
      htmlFor="beskrivelse"
    >
      <div
        className="flex label"
      >
        Beskrivelse
      </div>
    </label>
    <input
      className="skjemaelement__input input--fullbredde"
      id="beskrivelse"
      name="beskrivelse"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value=""
    />
    <div
      aria-live="assertive"
      role="alert"
    />
  </div>,
  <div
    className="skjemaelement"
  >
    <label
      className="skjemaelement__label"
      htmlFor="beloep"
    >
      <div
        className="flex label"
      >
        Beløp
      </div>
    </label>
    <input
      className="skjemaelement__input input--fullbredde"
      id="beloep"
      name="beloep"
      onBlur={[Function]}
      onChange={[Function]}
      type="number"
      value="0.00"
    />
    <div
      aria-live="assertive"
      role="alert"
    />
  </div>,
  <div
    className=""
    field={
      Object {
        "name": "nyInntektUtloeserArbeidsgiveravgift",
        "onBlur": [Function],
        "onChange": [Function],
        "value": undefined,
      }
    }
    form={
      Object {
        "dirty": false,
        "errors": Object {},
        "handleBlur": [Function],
        "handleChange": [Function],
        "handleReset": [Function],
        "handleSubmit": [Function],
        "initialValues": Object {
          "beloep": "0.00",
          "beskrivelse": "",
        },
        "isSubmitting": false,
        "isValid": false,
        "isValidating": false,
        "registerField": [Function],
        "resetForm": [Function],
        "setError": [Function],
        "setErrors": [Function],
        "setFieldError": [Function],
        "setFieldTouched": [Function],
        "setFieldValue": [Function],
        "setFormikState": [Function],
        "setStatus": [Function],
        "setSubmitting": [Function],
        "setTouched": [Function],
        "setValues": [Function],
        "status": undefined,
        "submitCount": 0,
        "submitForm": [Function],
        "touched": Object {},
        "unregisterField": [Function],
        "validateField": [Function],
        "validateForm": [Function],
        "validateOnBlur": true,
        "validateOnChange": true,
        "values": Object {
          "beloep": "0.00",
          "beskrivelse": "",
        },
      }
    }
  >
    <fieldset
      className="skjema__fieldset"
    >
      <legend
        className="skjema__legend"
      >
        Utløser arbeidsgiveravgift?
      </legend>
      <div
        className="skjemaelement skjemaelement--horisontal"
      >
        <input
          className="skjemaelement__input radioknapp"
          id="nyInntektJa"
          name="nyInntektUtloeserArbeidsgiveravgift"
          type="radio"
          value={true}
        />
        <label
          className="skjemaelement__label"
          htmlFor="nyInntektJa"
        >
          Ja
        </label>
      </div>
      <div
        className="skjemaelement skjemaelement--horisontal"
      >
        <input
          className="skjemaelement__input radioknapp"
          id="nyInntektNei"
          name="nyInntektUtloeserArbeidsgiveravgift"
          type="radio"
          value={false}
        />
        <label
          className="skjemaelement__label"
          htmlFor="nyInntektNei"
        >
          Nei
        </label>
      </div>
    </fieldset>
    <div
      aria-live="assertive"
      role="alert"
    />
  </div>,
  <button
    className="knapp knapp--hoved"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Legg til
  </button>,
  <button
    className="knapp knapp--standard"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Avbryt
  </button>,
]
`;
